numpy/core/tests/test_umath.py

```diff
@@ -732,6 +732,80 @@ def test_float_remainder_corner_cases(self):
                 assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, rem))
 
 
+class TestDivisionOverflows:
+    import operator
+
+    overflow_results = {
+        np.remainder: '0',
+        np.fmod: '0',
+        operator.mod: '0',
+        operator.floordiv: '-np.iinfo(dividend_dtype).min',
+        np.floor_divide: '-np.iinfo(dividend_dtype).min',
+        np.divmod: '(-np.iinfo(dividend_dtype).min, 0)'
+    }
+
+    @pytest.mark.parametrize("dividend_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("divisor_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("operation",
+            [np.remainder, np.fmod, np.divmod, np.floor_divide,
+             operator.mod, operator.floordiv])
+    @np.errstate(divide='raise')
+    def test_overflows(self, dividend_dtype, divisor_dtype, operation):
+        # SIMD tries to perform the operation on as many elements as possible
+        # that is a multiple of the register's size. We resort to the
+        # default implementation for the leftover elements.
+        # We try to cover all paths here.
+        arrays = [np.array([np.iinfo(dividend_dtype).min]*i,
+                           dtype=dividend_dtype) for i in range(1, 129)]
+        divisor = np.array([-1], dtype=divisor_dtype)
+        # If dividend is a larger type than the divisor (`else` case),
+        # then, result will be a larger type than dividend and will not
+        # result in an overflow for `divmod` and `floor_divide`.
+        if np.dtype(dividend_dtype).itemsize >= np.dtype(
+                divisor_dtype).itemsize and operation in (
+                        np.divmod, np.floor_divide,
+                        TestDivisionOverflows.operator.floordiv):
+            # FIXME: Raise the same error in case of scalars and arrays
+            with pytest.raises(
+                    RuntimeWarning if operation is
```

Comment:
Not very happy with this difference in behavior, but thought of following up as it might end up a bigger refactor? Ok with fixing now as well, let me know what's best.

Reply:
Subtleties...  Which errors are we giving?  `int_min / -1` (or all other operations), should probably give overflow warnings (rather than divide-by-zero), and I think this is what you see here?

But I am not sure if not all operations here should be overflow warnings/errors (which we should set to raise in the decorator, since we are testing them)?

Reply:
I think `RuntimeWarning` for underflow is the best error.

Now I think the reason for so much of different behaviors is because floor_div in SIMD is not handled in all loops. Your guess on
>  check whether we may have other SIMD paths that can trigger the issue. I assume not, and if we do the tests should weed them out.

is accurate. I'll fix all of them here and set the result to 0 making it uniform across.

I think we all can agree -MIN is definitely an overflow and lets change it to 0

Reply:
> I think we all can agree -MIN is definitely an overflow and lets change it to 0

But the correct "rollover" result of `-MIN` overflowing is `-MIN`?  So I think it is the better  result, especially for `-MIN` itself (and why would `MIN/-1` be different?).  `MIN` is much more of a warning that something is wrong and it seems also like the "well defined" behavior (if the C-standard would define 2-complement signed integer overflow as defined rather than undefined)

Reply:
Ah ok that does make sense. I forgot we do have a return value to indicate any errors as well. I'll go ahead and set it to `-MIN` (in a way the compiler does not treat it as a warning)

Reply:
Yeah, we don't have it really, I guess.  But I do feel it may be useful to users to get `MIN`; 0 is just not a "weird" value :).

Reply:
Haha yeah agree on that. Yeah let's set it to `MIN` and raise an `Overflow`.

Reply:
Just to summarize how I current see things:
* For the "overflow" paths, I would argue for `MIN` as the correct result and would be happy if we change it if there are currently paths that return 0.
* For the "division by zero" paths, I don't have a clear opinion.  To some degree, I think `MIN` might be the better "error" value, but I think we had been consistent about returning 0, so I tend to think we should just keep doing that.

Reply:
+1 to the proposal, I'll make the changes following this. It would be nice If we can officially standardize it somewhere :)